symbolic-debuginfo/src/sourcebundle.rs

```diff
@@ -858,7 +865,7 @@ where
     pub fn write_object<O>(self, object: &O, object_name: &str) -> Result<bool, SourceBundleError>
     where
         O: ObjectLike,
-        O::Error: Fail,
+        O::Error: Into<ObjectError>,
```

Comment:
I'm realizing that this is fairly restrictive now, essentially making it impossible that anyone outside of symbolic implements `ObjectLike` and then uses it with this class. I think we'll have to go back to `Box<dyn>` for this, or we simply swallow the cause.

Reply:
note to self: The `O::Error` constraint should be `dyn Error + Send + Sync + 'static`, and in the places where we save it its boxed.